toxcore/TCP_client.c

```diff
@@ -183,6 +193,28 @@ static void proxy_socks5_generate_connection_request(TCP_Client_Connection *TCP_
     TCP_conn->last_packet_sent = 0;
 }
 
+static void proxy_socks5_generate_auth_request(TCP_Client_Connection *TCP_conn)
+{
+    uint8_t username_len = strlen(TCP_conn->proxy_info.username);
+    uint8_t password_len = strlen(TCP_conn->proxy_info.password);
```

Comment:
you can use string functions in tox.c but not here.

Save the length in the proxy settings struct.